server/belga/publish/belga_newsml_1_2.py

```diff
@@ -864,55 +953,73 @@ def _format_newslines(self, newscomponent_2_level, item):
         :param Element newscomponent_2_level: NewsComponent of 2nd level
         :param dict item: item
         """
-        newslines = SubElement(newscomponent_2_level, 'NewsLines')
-        SubElement(newslines, 'DateLine')
-        SubElement(newslines, 'HeadLine').text = item.get('headline')
-        SubElement(newslines, 'CopyrightLine').text = item.get('copyrightholder')
-        SubElement(newslines, 'CreditLine').text = self.DEFAULT_CREDITLINE
+        newslines = SubElement(newscomponent_2_level, "NewsLines")
+        SubElement(newslines, "DateLine")
+        SubElement(newslines, "HeadLine").text = item.get("headline")
+        SubElement(newslines, "CopyrightLine").text = item.get("copyrightholder")
+
+        # SDBELGA-672
+        for subject in item.get("subject", []):
+            if subject.get("scheme") == "sources":
+                SubElement(newslines, "CreditLine").text = subject.get("name")
+
+        if len(newslines.xpath("CreditLine")) == 0:
```

Comment:
imo this might be a bit more readable if it would be like:
```python
for subject in item.get("subject", []):
    if subject.get("scheme") == "sources":
        SubElement(newslines, "CreditLine").text = subject.get("name")
        break
else:
    if item.get("source"):
        ...
```
so it's more obvious those blocks belong together